libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -1204,6 +1222,99 @@ async def ainvoke(
             await run_manager.on_chain_end(output)
             return output
 
+    def stream(
+        self, input: Input, config: Optional[RunnableConfig] = None
+    ) -> Iterator[RunnableMapChunk]:
+        from langchain.callbacks.manager import CallbackManager
+
+        # setup callbacks
+        config = config or {}
+        callback_manager = CallbackManager.configure(
+            inheritable_callbacks=config.get("callbacks"),
+            local_callbacks=None,
+            verbose=False,
+            inheritable_tags=config.get("tags"),
+            local_tags=None,
+            inheritable_metadata=config.get("metadata"),
+            local_metadata=None,
+        )
+        # start the root run
+        run_manager = callback_manager.on_chain_start(
+            dumpd(self), input if isinstance(input, dict) else {"input": input}
+        )
+
+        steps = dict(self.steps)
+        final_output = None
+        try:
+            for results in zip_longest(
+                *(
+                    step.stream(input, patch_config(config, run_manager.get_child()))
```

Comment:
This is doing them in sequence, we need to use ThreadPoolExecutor to run each step in parallel

Reply:
It pads iterators with no chunks remaining with `None`

Reply:
Will have a look at it today

Reply:
Same issue in the async one I think, but solution will be different